test/common/stats/thread_local_store_test.cc

```diff
@@ -80,12 +80,38 @@ class StatsThreadLocalStoreTest : public testing::Test, public RawStatDataAlloca
     store_->addSink(sink_);
   }
 
-  void setUp() {
+  MOCK_METHOD1(alloc, RawStatData*(const std::string& name));
+  MOCK_METHOD1(free, void(RawStatData& data));
+
+  NiceMock<Event::MockDispatcher> main_thread_dispatcher_;
+  NiceMock<ThreadLocal::MockInstance> tls_;
+  TestAllocator alloc_;
+  MockSink sink_;
+  std::unique_ptr<ThreadLocalStoreImpl> store_;
+};
+
+class HistogramTest : public testing::Test, public RawStatDataAllocator {
+public:
+  HistogramTest() {
+    ON_CALL(*this, alloc(_)).WillByDefault(Invoke([this](const std::string& name) -> RawStatData* {
```

Comment:
Remove the ctor and move these lines to SetUp().

Reply:
done